test/HttpTestHelpers.h

```diff
@@ -35,15 +39,26 @@ class TestHttpServer {
   // Indicator whether the server has been shut down. We need this because
   // `HttpServer::shutDown` must only be called once.
   std::atomic<bool> hasBeenShutDown_ = false;
+  ad_utility::websocket::QueryRegistry registry_;
 
  public:
   // Create server on localhost. Port 0 instructs the operating system to choose
   // a free port of its choice.
   explicit TestHttpServer(HttpHandler httpHandler) {
     const std::string& ipAddress = "0.0.0.0";
     int numServerThreads = 1;
-    server_ = std::make_shared<HttpServer<HttpHandler>>(
-        0, ipAddress, numServerThreads, std::move(httpHandler));
+    auto webSocketSessionSupplier = [this](net::io_context& ioContext) {
+      ad_utility::websocket::QueryHub queryHub{ioContext};
+      return [this, queryHub = std::move(queryHub)](
+                 const http::request<http::string_body>& request,
```

Comment:
Yes and no, the tests have this problem because the `QueryHub` is created inside of a worker attached to the io context it's using, but since this instance is created outside of a worker it isn't affected by this problem.